browser/ui/webui/brave_shields/shields_data_handler.cc

```diff
@@ -49,6 +48,21 @@ void ShieldsDataHandler::GetSiteBlockInfo(GetSiteBlockInfoCallback callback) {
   std::move(callback).Run(site_block_info_.Clone());
 }
 
+void ShieldsDataHandler::GetAdBlockMode(GetAdBlockModeCallback callback) {
+  auto* shields_data_ctrlr = GetActiveShieldsDataController();
+  DCHECK(shields_data_ctrlr);
+
+  std::move(callback).Run(shields_data_ctrlr->GetAdBlockMode());
+}
+
+void ShieldsDataHandler::SetAdBlockMode(AdBlockMode mode) {
+  auto* shields_data_ctrlr = GetActiveShieldsDataController();
+  if (!shields_data_ctrlr)
```

Comment:
I think this should also be a DCHECK as above, especially since this method doesn't have a callback to notify that it potentially failed here

Reply:
added